bpf/lib/trace_sock.h

```diff
@@ -0,0 +1,120 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/* Copyright Authors of Cilium */
+
+/*
+ * Socket based service load-balancing notification via perf event ring buffer.
+ *
+ * API:
+ * void send_trace_sock_notify(ctx, obs_point, src_ip, dst_ip, src_port,
+ * 			       dst_port, sock_cookie, xlate_point, l4_proto)
+ *
+ * @ctx:	 socket address buffer
+ * @obs_point:	 observation point (TRACE_*)
+ * @dst_ip:	 destination ip address
+ * @dst_port:	 destination port
+ * @cgroup_id:   pod or container cgroup_id
+ * @sock_cookie: socket cookie
+ * @xlate_point: service translation point for load-balancing
+ * @l4_proto:    layer 4 protocol
+ *
+ * If TRACE_SOCK_NOTIFY is not defined, the API will be compiled in as a NOP.
+ */
+#ifndef __LIB_TRACE_SOCK__
+#define __LIB_TRACE_SOCK__
+
+#include <bpf/ctx/sock.h>
+
+#include "common.h"
+#include "events.h"
+
+/* L4 protocol for the trace event */
+enum l4_protocol {
+	L4_PROTOCOL_UNKNOWN = 0,
+	L4_PROTOCOL_TCP = 1,
+  	L4_PROTOCOL_UDP = 2,
+} __packed;
+
+/* Direction for translation between service and backend IP */
+enum xlate_point {
+	XLATE_PRE_DIRECTION_FWD = 0,  /* Pre service translation */
```

Comment:
Can we start this at offset 1? Similarly to `L4_PROTOCOL_UNKNOWN`. Having a zero-value for "UNKNOWN" simplifies the Hubble API, as we can assume value 0 for everything which is not source from `trace_sock_notify`.